File: Experiment Control/R21/NetComDevelopmentPackage_v3.1.0/NetComExamples/C#/NetComExampleCommands/NetComExampleCommandsForm.cs
//**********************************************
//	NetComExampleCommandsForm.cs
//	Copyright 2016 @ Neuralynx, Inc
//**********************************************
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

namespace NetComExampleCommands
{
	/// <summary>
	/// Summary description for Form1.
	/// </summary>
	public class NetComExampleCommandsForm : System.Windows.Forms.Form
	{
		#region UI Members
        internal System.Windows.Forms.Panel pnlCommands;
			internal System.Windows.Forms.Label lblCommandList;
			internal System.Windows.Forms.Button btnSendCommand;
			internal System.Windows.Forms.ComboBox cmbCommandList;
			internal System.Windows.Forms.Label lblServerNameComment;
			internal System.Windows.Forms.TextBox txtServerName;
			internal System.Windows.Forms.Button btnDisconnect;
			internal System.Windows.Forms.Button btnConnect;
			internal System.Windows.Forms.Label lblServerName;
		#endregion

		#region Private Members
			private MNetCom.MNetComClient mNetComClient;
		#endregion
            private Label lblReplyStringValue;
            private Label lblReplyStringLabel;


		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		#region Constructor
			public NetComExampleCommandsForm()
			{
				//initialize class vars
				this.mNetComClient = new MNetCom.MNetComClient();

				//
				// Required for Windows Form Designer support
				//
				InitializeComponent();

				//
				// TODO: Add any constructor code after InitializeComponent call
				//

				//extra UI setup
				this.cmbCommandList.SelectedIndex = 0;
				this.btnDisconnect.Enabled = false;
				this.pnlCommands.Enabled = false;
				
				//Set the logfile name
				if (!(this.mNetComClient.SetLogFileName(Application.StartupPath.ToString() + "\\NetComExampleCommandsLogfile.txt")))
				{
					   MessageBox.Show(this, "Call to set the logfile name failed", "NetCom Error", MessageBoxButtons.OK, MessageBoxIcon.Error, MessageBoxDefaultButton.Button1);
				}

			}
		#endregion


		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if (components != null) 
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
            this.pnlCommands = new System.Windows.Forms.Panel();
            this.lblCommandList = new System.Windows.Forms.Label();
            this.btnSendCommand = new System.Windows.Forms.Button();
            this.cmbCommandList = new System.Windows.Forms.ComboBox();
            this.lblServerNameComment = new System.Windows.Forms.Label();
            this.txtServerName = new System.Windows.Forms.TextBox();
            this.btnDisconnect = new System.Windows.Forms.Button();
            this.btnConnect = new System.Windows.Forms.Button();
            this.lblServerName = new System.Windows.Forms.Label();
            this.lblReplyStringLabel = new System.Windows.Forms.Label();
            this.lblReplyStringValue = new System.Windows.Forms.Label();
            this.pnlCommands.SuspendLayout();
            this.SuspendLayout();
            // 
            // pnlCommands
            // 
            this.pnlCommands.Controls.Add(this.lblReplyStringValue);
            this.pnlCommands.Controls.Add(this.lblReplyStringLabel);
            this.pnlCommands.Controls.Add(this.lblCommandList);
            this.pnlCommands.Controls.Add(this.btnSendCommand);
            this.pnlCommands.Controls.Add(this.cmbCommandList);
            this.pnlCommands.Location = new System.Drawing.Point(16, 64);
            this.pnlCommands.Name = "pnlCommands";
            this.pnlCommands.Size = new System.Drawing.Size(488, 65);
            this.pnlCommands.TabIndex = 12;
            // 
            // lblCommandList
            // 
            this.lblCommandList.Location = new System.Drawing.Point(-3, 16);
            this.lblCommandList.Name = "lblCommandList";
            this.lblCommandList.Size = new System.Drawing.Size(93, 21);
            this.lblCommandList.TabIndex = 24;
            this.lblCommandList.Text = "Command List";
            this.lblCommandList.TextAlign = System.Drawing.ContentAlignment.MiddleRight;
            // 
            // btnSendCommand
            // 
            this.btnSendCommand.Location = new System.Drawing.Point(392, 17);
            this.btnSendCommand.Name = "btnSendCommand";
            this.btnSendCommand.Size = new System.Drawing.Size(96, 23);
            this.btnSendCommand.TabIndex = 23;
            this.btnSendCommand.Text = "Send Command";
            this.btnSendCommand.Click += new System.EventHandler(this.btnSendCommand_Click);
            // 
            // cmbCommandList
            // 
            this.cmbCommandList.Items.AddRange(new object[] {
            "-PostEvent \"Test Event\" 0 0",
            "-StartAcquisition"});
            this.cmbCommandList.Location = new System.Drawing.Point(96, 17);
            this.cmbCommandList.Name = "cmbCommandList";
            this.cmbCommandList.Size = new System.Drawing.Size(288, 21);
            this.cmbCommandList.TabIndex = 22;
            // 
            // lblServerNameComment
            // 
            this.lblServerNameComment.Location = new System.Drawing.Point(112, 40);
            this.lblServerNameComment.Name = "lblServerNameComment";
            this.lblServerNameComment.Size = new System.Drawing.Size(208, 16);
            this.lblServerNameComment.TabIndex = 11;
            this.lblServerNameComment.Text = "(User may enter pc name or IP address)";
            // 
            // txtServerName
            // 
            this.txtServerName.Location = new System.Drawing.Point(96, 16);
            this.txtServerName.Name = "txtServerName";
            this.txtServerName.Size = new System.Drawing.Size(232, 20);
            this.txtServerName.TabIndex = 10;
            // 
            // btnDisconnect
            // 
            this.btnDisconnect.Location = new System.Drawing.Point(432, 16);
            this.btnDisconnect.Name = "btnDisconnect";
            this.btnDisconnect.Size = new System.Drawing.Size(75, 23);
            this.btnDisconnect.TabIndex = 9;
            this.btnDisconnect.Text = "Disconnect";
            this.btnDisconnect.Click += new System.EventHandler(this.btnDisconnect_Click);
            // 
            // btnConnect
            // 
            this.btnConnect.Location = new System.Drawing.Point(344, 16);
            this.btnConnect.Name = "btnConnect";
            this.btnConnect.Size = new System.Drawing.Size(75, 23);
            this.btnConnect.TabIndex = 8;
            this.btnConnect.Text = "Connect";
            this.btnConnect.Click += new System.EventHandler(this.btnConnect_Click);
            // 
            // lblServerName
            // 
            this.lblServerName.Location = new System.Drawing.Point(16, 16);
            this.lblServerName.Name = "lblServerName";
            this.lblServerName.Size = new System.Drawing.Size(72, 16);
            this.lblServerName.TabIndex = 7;
            this.lblServerName.Text = "Server Name";
            // 
            // lblReplyStringLabel
            // 
            this.lblReplyStringLabel.AutoSize = true;
            this.lblReplyStringLabel.Location = new System.Drawing.Point(26, 45);
            this.lblReplyStringLabel.Name = "lblReplyStringLabel";
            this.lblReplyStringLabel.Size = new System.Drawing.Size(64, 13);
            this.lblReplyStringLabel.TabIndex = 25;
            this.lblReplyStringLabel.Text = "Reply String";
            this.lblReplyStringLabel.TextAlign = System.Drawing.ContentAlignment.MiddleRight;
            // 
            // lblReplyStringValue
            // 
            this.lblReplyStringValue.AutoSize = true;
            this.lblReplyStringValue.Location = new System.Drawing.Point(96, 45);
            this.lblReplyStringValue.Name = "lblReplyStringValue";
            this.lblReplyStringValue.Size = new System.Drawing.Size(10, 13);
            this.lblReplyStringValue.TabIndex = 26;
            this.lblReplyStringValue.Text = " ";
            this.lblReplyStringValue.TextAlign = System.Drawing.ContentAlignment.MiddleLeft;
            // 
            // NetComExampleCommandsForm
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.ClientSize = new System.Drawing.Size(520, 133);
            this.Controls.Add(this.pnlCommands);
            this.Controls.Add(this.lblServerNameComment);
            this.Controls.Add(this.txtServerName);
            this.Controls.Add(this.btnDisconnect);
            this.Controls.Add(this.btnConnect);
            this.Controls.Add(this.lblServerName);
            this.Name = "NetComExampleCommandsForm";
            this.Text = "NetCom Examples - Commands";
            this.pnlCommands.ResumeLayout(false);
            this.pnlCommands.PerformLayout();
            this.ResumeLayout(false);
            this.PerformLayout();

		}
		#endregion

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Application.Run(new NetComExampleCommandsForm());
		}

		#region UI Event Handlers
		//**************************************************************
		//**************************************************************
		private void btnConnect_Click(object sender, System.EventArgs e)
		{
			if (!( this.mNetComClient.AreWeConnected()))
			{
				if (this.mNetComClient.ConnectToServer(this.txtServerName.Text))
				{
					this.mNetComClient.SetApplicationName("NetCom Commands Example");

					//change ui to show connected status
					this.txtServerName.Enabled = false;
					this.btnConnect.Enabled = false;
					this.btnDisconnect.Enabled = true;
					this.pnlCommands.Enabled = true;
				}
				else
				{
					MessageBox.Show(this, "Connection to server failed", "NetCom Error", MessageBoxButtons.OK, MessageBoxIcon.Error, MessageBoxDefaultButton.Button1);
				}
			}
		}

		//**************************************************************
		//**************************************************************
		private void btnDisconnect_Click(object sender, System.EventArgs e)
		{
			if (this.mNetComClient.AreWeConnected())
			{
				if ( this.mNetComClient.DisconnectFromServer() )
				{
					//change ui to show disconnected status
					this.txtServerName.Enabled = true;
					this.btnConnect.Enabled = true;
					this.btnDisconnect.Enabled = false;
					this.pnlCommands.Enabled = false;
				}
				else
				{
					MessageBox.Show(this, "Disconnection from server failed", "NetCom Error", MessageBoxButtons.OK, MessageBoxIcon.Error, MessageBoxDefaultButton.Button1);
				}
			}
		}

		//**************************************************************
		//**************************************************************
		private void btnSendCommand_Click(object sender, System.EventArgs e)
		{
			string reply = "";
			if (!(mNetComClient.SendCommand(this.cmbCommandList.Text, ref reply)))
			{
				MessageBox.Show(this, "Send command to server failed", "NetCom Error", MessageBoxButtons.OK, MessageBoxIcon.Error, MessageBoxDefaultButton.Button1);
			}else {
                String[] parsedReplyString = reply.Split(' ');
                if (0 < parsedReplyString.GetLength(0)) {
                    if( parsedReplyString[0].Equals( "-1" )){
                        MessageBox.Show(this, "Cheetah could not process your command.", "NetCom Error", MessageBoxButtons.OK, MessageBoxIcon.Error, MessageBoxDefaultButton.Button1);
                    }
                }
                this.lblReplyStringValue.Text = reply;
            }
		
		}

		#endregion

	}
}

File: Experiment Control/R21/NetComDevelopmentPackage_v3.1.0/NetComExamples/VB.NET/NetComExampleCommands/NetComExampleCommandsForm.vb
Public Class NetComExampleCommands
    Inherits System.Windows.Forms.Form
#Region "Private Members"
    Private mNetComClient As MNetCom.MNetComClient
#End Region

#Region "Constructor"
    Public Sub New()
        MyBase.New()

        'initialize class vars
        mNetComClient = New MNetCom.MNetComClient


        'This call is required by the Windows Form Designer.
        InitializeComponent()

        'Add any initialization after the InitializeComponent() call

        'extra UI setup
        cmbCommandList.SelectedIndex = 0
        btnDisconnect.Enabled = False
        pnlCommands.Enabled = False

        'Set the logfile name
        If Not mNetComClient.SetLogFileName(Application.StartupPath.ToString() & "\NetComExampleCommandsLogfile.txt") Then
            MessageBox.Show(Me, "Call to set the logfile name failed", "NetCom Error", MessageBoxButtons.OK, MessageBoxIcon.Error, MessageBoxDefaultButton.Button1)
        End If

    End Sub
#End Region

#Region " Windows Form Designer generated code "



    'Form overrides dispose to clean up the component list.
    Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
        If disposing Then
            If Not (components Is Nothing) Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)
    End Sub

    'Required by the Windows Form Designer
    Private components As System.ComponentModel.IContainer

    'NOTE: The following procedure is required by the Windows Form Designer
    'It can be modified using the Windows Form Designer.  
    'Do not modify it using the code editor.
    Friend WithEvents lblServerName As System.Windows.Forms.Label
    Friend WithEvents btnConnect As System.Windows.Forms.Button
    Friend WithEvents btnDisconnect As System.Windows.Forms.Button
    Friend WithEvents txtServerName As System.Windows.Forms.TextBox
    Friend WithEvents lblServerNameComment As System.Windows.Forms.Label
    Friend WithEvents pnlCommands As System.Windows.Forms.Panel
    Friend WithEvents lblCommandListLabel As System.Windows.Forms.Label
    Friend WithEvents btnSendCommand As System.Windows.Forms.Button
    Friend WithEvents lblReplyStringValue As System.Windows.Forms.Label
    Friend WithEvents lblReplyStringLabel As System.Windows.Forms.Label
    Friend WithEvents cmbCommandList As System.Windows.Forms.ComboBox
    <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
        lblServerName = New System.Windows.Forms.Label
        btnConnect = New System.Windows.Forms.Button
        btnDisconnect = New System.Windows.Forms.Button
        txtServerName = New System.Windows.Forms.TextBox
        lblServerNameComment = New System.Windows.Forms.Label
        pnlCommands = New System.Windows.Forms.Panel
        lblCommandListLabel = New System.Windows.Forms.Label
        btnSendCommand = New System.Windows.Forms.Button
        cmbCommandList = New System.Windows.Forms.ComboBox
        lblReplyStringLabel = New System.Windows.Forms.Label
        lblReplyStringValue = New System.Windows.Forms.Label
        pnlCommands.SuspendLayout()
        SuspendLayout()
        '
        'lblServerName
        '
        lblServerName.Location = New System.Drawing.Point(16, 11)
        lblServerName.Name = "lblServerName"
        lblServerName.Size = New System.Drawing.Size(72, 16)
        lblServerName.TabIndex = 0
        lblServerName.Text = "Server Name"
        '
        'btnConnect
        '
        btnConnect.Location = New System.Drawing.Point(344, 8)
        btnConnect.Name = "btnConnect"
        btnConnect.Size = New System.Drawing.Size(75, 23)
        btnConnect.TabIndex = 1
        btnConnect.Text = "Connect"
        '
        'btnDisconnect
        '
        btnDisconnect.Location = New System.Drawing.Point(432, 8)
        btnDisconnect.Name = "btnDisconnect"
        btnDisconnect.Size = New System.Drawing.Size(75, 23)
        btnDisconnect.TabIndex = 2
        btnDisconnect.Text = "Disconnect"
        '
        'txtServerName
        '
        txtServerName.Location = New System.Drawing.Point(96, 9)
        txtServerName.Name = "txtServerName"
        txtServerName.Size = New System.Drawing.Size(232, 20)
        txtServerName.TabIndex = 3
        '
        'lblServerNameComment
        '
        lblServerNameComment.Location = New System.Drawing.Point(108, 32)
        lblServerNameComment.Name = "lblServerNameComment"
        lblServerNameComment.Size = New System.Drawing.Size(208, 16)
        lblServerNameComment.TabIndex = 5
        lblServerNameComment.Text = "(User may enter pc name or IP address)"
        '
        'pnlCommands
        '
        pnlCommands.Anchor = CType(((System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Left) _
                    Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles)
        pnlCommands.Controls.Add(lblReplyStringValue)
        pnlCommands.Controls.Add(lblReplyStringLabel)
        pnlCommands.Controls.Add(lblCommandListLabel)
        pnlCommands.Controls.Add(btnSendCommand)
        pnlCommands.Controls.Add(cmbCommandList)
        pnlCommands.Location = New System.Drawing.Point(16, 57)
        pnlCommands.Name = "pnlCommands"
        pnlCommands.Size = New System.Drawing.Size(488, 70)
        pnlCommands.TabIndex = 6
        '
        'lblCommandListLabel
        '
        lblCommandListLabel.Location = New System.Drawing.Point(13, 17)
        lblCommandListLabel.Name = "lblCommandListLabel"
        lblCommandListLabel.Size = New System.Drawing.Size(77, 20)
        lblCommandListLabel.TabIndex = 24
        lblCommandListLabel.Text = "Command List"
        lblCommandListLabel.TextAlign = System.Drawing.ContentAlignment.MiddleRight
        '
        'btnSendCommand
        '
        btnSendCommand.Location = New System.Drawing.Point(392, 17)
        btnSendCommand.Name = "btnSendCommand"
        btnSendCommand.Size = New System.Drawing.Size(96, 23)
        btnSendCommand.TabIndex = 23
        btnSendCommand.Text = "Send Command"
        '
        'cmbCommandList
        '
        cmbCommandList.Items.AddRange(New Object() {"-PostEvent ""Test Event"" 0 0", "-StartAcquisition"})
        cmbCommandList.Location = New System.Drawing.Point(96, 17)
        cmbCommandList.Name = "cmbCommandList"
        cmbCommandList.Size = New System.Drawing.Size(288, 21)
        cmbCommandList.TabIndex = 22
        '
        'lblReplyStringLabel
        '
        lblReplyStringLabel.AutoSize = True
        lblReplyStringLabel.Location = New System.Drawing.Point(26, 47)
        lblReplyStringLabel.Name = "lblReplyStringLabel"
        lblReplyStringLabel.Size = New System.Drawing.Size(64, 13)
        lblReplyStringLabel.TabIndex = 25
        lblReplyStringLabel.Text = "Reply String"
        '
        'lblReplyStringValue
        '
        lblReplyStringValue.AutoSize = True
        lblReplyStringValue.Location = New System.Drawing.Point(95, 47)
        lblReplyStringValue.Name = "lblReplyStringValue"
        lblReplyStringValue.Size = New System.Drawing.Size(0, 13)
        lblReplyStringValue.TabIndex = 26
        lblReplyStringValue.TextAlign = System.Drawing.ContentAlignment.MiddleLeft
        '
        'NetComExampleCommands
        '
        AutoScaleBaseSize = New System.Drawing.Size(5, 13)
        ClientSize = New System.Drawing.Size(520, 139)
        Controls.Add(pnlCommands)
        Controls.Add(lblServerNameComment)
        Controls.Add(txtServerName)
        Controls.Add(btnDisconnect)
        Controls.Add(btnConnect)
        Controls.Add(lblServerName)
        FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog
        Name = "NetComExampleCommands"
        Text = "NetCom Example - Commands"
        pnlCommands.ResumeLayout(False)
        pnlCommands.PerformLayout()
        ResumeLayout(False)
        PerformLayout()

    End Sub

#End Region


#Region "UI Event Handlers"

    '****************************************************************************************************************
    '****************************************************************************************************************
    Private Sub btnConnect_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnConnect.Click
        If Not mNetComClient.AreWeConnected Then
            If mNetComClient.ConnectToServer(txtServerName.Text) Then
                mNetComClient.SetApplicationName("NetCom Commands Example")

                'change ui to show connected status
                txtServerName.Enabled = False
                btnConnect.Enabled = False
                btnDisconnect.Enabled = True
                pnlCommands.Enabled = True

            Else
                MessageBox.Show(Me, "Connection to server failed", "NetCom Error", MessageBoxButtons.OK, MessageBoxIcon.Error, MessageBoxDefaultButton.Button1)
            End If
        End If
    End Sub

    '****************************************************************************************************************
    '****************************************************************************************************************
    Private Sub btnDisconnect_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnDisconnect.Click
        If mNetComClient.AreWeConnected Then
            If mNetComClient.DisconnectFromServer Then
                'change ui to show disconnected status
                txtServerName.Enabled = True
                btnConnect.Enabled = True
                btnDisconnect.Enabled = False
                pnlCommands.Enabled = False
            Else
                MessageBox.Show(Me, "Disconnection from server failed", "NetCom Error", MessageBoxButtons.OK, MessageBoxIcon.Error, MessageBoxDefaultButton.Button1)
            End If
        End If
    End Sub

    '****************************************************************************************************************
    '****************************************************************************************************************
    Private Sub btnSendCommand_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSendCommand.Click
        Dim reply As String = ""
        Dim cmd As String = cmbCommandList.Text
        If Not mNetComClient.SendCommand(cmbCommandList.Text, reply) Then
            MessageBox.Show(Me, "Send command to server failed", "NetCom Error", MessageBoxButtons.OK, MessageBoxIcon.Error, MessageBoxDefaultButton.Button1)
        Else
            Dim parsedReplyString As String() = Strings.Split(reply, " ")
            If (0 < parsedReplyString.GetLength(0)) Then
                If parsedReplyString(0) = "-1" Then
                    MessageBox.Show(Me, "The DAS could not process your command.", "NetCom Error", MessageBoxButtons.OK, MessageBoxIcon.Error, MessageBoxDefaultButton.Button1)
                End If

                lblReplyStringValue.Text = reply

            End If

        End If
    End Sub

#End Region

End Class
